src/libclipper/src/persistent_state.cpp

```diff
@@ -42,21 +48,58 @@ StateDB::StateDB() {
 StateDB::~StateDB() { redis_connection_.disconnect(); }
 
 boost::optional<std::string> StateDB::get(const StateKey& key) {
-  std::string redis_key = generate_redis_key(key);
-  const std::vector<std::string> cmd_vec{"GET", redis_key};
-  return redis::send_cmd_with_reply<std::string>(redis_connection_, cmd_vec);
+  auto entry_search = cache_.find(key);
+  if (entry_search != cache_.end()) {
+    auto& cache_entry = entry_search->second;
+    std::lock_guard<std::mutex> entry_lock(cache_entry->entry_mtx_);
+    return cache_entry->entry_value_;
+  } else {
+    std::string redis_key = generate_redis_key(key);
+    const std::vector<std::string> cmd_vec{"GET", redis_key};
+    auto result =
+        redis::send_cmd_with_reply<std::string>(redis_connection_, cmd_vec);
```

Comment:
If you fetch the value from Redis, you should cache it here as well.

Reply:
Done